contrastsecurity/README.md

```diff
@@ -0,0 +1,64 @@
+# Contrast Security Integration
+
+## Overview
+
+The Datadog-Constrant integration allows you to get your Contrast logs into Datadog.
+
+## Setup
+
+### Set up Contrast Protect logs collection
+
+Enable logs collection for Datadog Agent in `/etc/datadog-agent/datadog.yaml` on Linux platforms. On other platforms, refer to the [Agent Configuration Files guide](https://docs.datadoghq.com/agent/guide/agent-configuration-files/?tab=agentv6) for the location of your configuration file:
+```
+logs_enabled: true
+```
+
+* Create a new folder `java.d` in the `conf.d/` directory of DataDog configuration directory.
+* Create a new `conf.yaml` file.
+* Add a custom log collection configuration group.
+```
+logs:
+  - type: file
+    path: /path/to/contrast/security.log
+    service: contrast
+    source: java
+```
+
+* [Restart the Datadog Agent](https://docs.datadoghq.com/agent/guide/agent-commands/?tab=agentv6#restart-the-agent).
+
+For more information, consult the [Logs Collection documentation](https://docs.datadoghq.com/logs/log_collection/?tab=tailexistingfiles#getting-started-with-the-agent).
+
+### Create a Logs Processing Pipeline
+
+* Create a new pipeline in Datadog (*Logs* -> *Configuration* - > *New Pipeline*) 
+* Expand the new pipeline and click *Add Processor*.
+* Add a Grok parser to the pipeline with the following parsing rule:
+```
+ContrastSecurityLogRule %{data:data}pri=%{data:pri} src=%{ip:src} spt=%{number:spt} request=%{data:request} requestMethod=%{word:requestMethod} app=%{data:app} outcome=%{word:outcome}
+```
+
```

Comment:
Thanks for sharing this.

It seems that we can definitely help to provide a more optimised parsing rule (see [keyvalue examples](https://docs.datadoghq.com/logs/processing/parsing/?tab=matcher#key-value)) that would match the global Datadog naming convention so that attribute name would match the same as other integration (`http.method` instead of requestMethod` for instance).

So what we can do there is to create on Datadog side (still not open source but we are working on it) an integration pipeline that would get installed automatically.
For this to happen, would you mind sharing a couple log samples?

Then you will be able to simply remove this part and the integration pipeline would be automatically installed when sending logs with the `contrastsecurity` source.

Reply:
2019-11-11 03:17:58,857 [Contrast Server Activity Updater wire] DEBUG -  << "HTTP/1.1 304 Not Modified[\r][\n]"
2019-11-11 03:17:58,857 [Contrast Server Activity Updater wire] DEBUG -  << "Server: Apache-Coyote/1.1[\r][\n]"
2019-11-11 03:17:58,857 [Contrast Server Activity Updater wire] DEBUG -  << "Pragma: no-cache[\r][\n]"
2019-11-11 03:17:58,857 [Contrast Server Activity Updater wire] DEBUG -  << "Expires: Thu, 01 Jan 1970 00:00:00 GMT[\r][\n]"
2019-11-11 03:17:58,857 [Contrast Server Activity Updater wire] DEBUG -  << "Cache-Control: no-cache, no-store[\r][\n]"
2019-11-11 03:17:58,857 [Contrast Server Activity Updater wire] DEBUG -  << "X-Frame-Options: SAMEORIGIN[\r][\n]"
2019-11-11 03:17:58,857 [Contrast Server Activity Updater wire] DEBUG -  << "X-XSS-Protection: 1;mode=block[\r][\n]"
2019-11-11 03:17:58,857 [Contrast Server Activity Updater wire] DEBUG -  << "X-Content-Type-Options: nosniff[\r][\n]"
2019-11-11 03:17:58,857 [Contrast Server Activity Updater wire] DEBUG -  << "Date: Mon, 11 Nov 2019 08:17:58 GMT[\r][\n]"
2019-11-11 03:17:58,857 [Contrast Server Activity Updater wire] DEBUG -  << "[\r][\n]"



Are these suitable example?

Reply:
Thanks for this.

Just wondering if there are other types because when l look at the rule you suggested there are some `request` and `requestMethod` that I cannot see the in the samples.

Reply:
No. there is not going to be anything more specific request/requestmethod

Reply:
Therefore may I ask where the rules you suggested initially are coming for? For reference:

```
ContrastSecurityLogRule %{data:data}pri=%{data:pri} src=%{ip:src} spt=%{number:spt} request=%{data:request} requestMethod=%{word:requestMethod} app=%{data:app} outcome=%{word:outcome}
```

I just want to make sure we do not miss any interested log formats there.

Reply:
A co-worker created this rule. I would be open to editing it as needed to whatever it needs to be.

Reply:
Do you know if you could ask this co-worker which logs were used to build this rule?
As I would expect something like:

```
pri=123 src=127.0.0.1 spt=12 request=https://example.com requestMethod=GET app=myapp outcome=success
```

which seems very different from the samples that you shared. So I'm thinking that we likely want to support both format so having examples for this format as well from your co-worker could be super helpful.

Reply:
``[INFO] [talledLocalContainer] 261119 13.03.21,597 {} {} {} INFO  (VulnerabilityProcessorQueryService.java:135) [VULNERABILITIES] Saved vulnerability instance property VulnerabilityInstanceProperty(id=19, vulnerabilityInstanceId=0, hash=1805407380, vulnerabilityHash=1805407380, name=/WebGoat/SqlInjection.lesson.lesson, value={"html":"class\u003d\"assignment-success\"\u003e\u003ci class\u003d\"fa fa-2 fa-check hidden\" aria-hidden\u003d\"true\"\u003e\u003c/i\u003e\u003c/div\u003e\n        \u003cform class\u003d\"attack-form\" accept-charset\u003d\"UNKNOWN\" method\u003d\"POST\" name\u003d\"form\" action\u003d\"/WebGoat/SqlInjection/attack5a\" enctype\u003d\"application/json;charset\u003dUTF-8\"\u003e\n            \u003ctable\u003e\n                \u003ctr\u003e\n                    \u003ctd\u003eAccount Name:\u003c/td\u003e\n               ","startIdx":100,"endIdx":257}, type=EVIDENCE)``` is another example log. But I am willing to change for whatever is advised!

Reply:
Thanks for this, that's super helpful.

When you say that you are willing to change for whatever is advised, are you saying that you can change the default logging format for contrast security logs?

As here it seems that the `value` part of the logs are encoded (see the `\u003c`) blocks and on our end it would be super easy to parse this if they did not had those encoded blocks.
Is that something that can be changed on your end?

Reply:
oh I am willing to change the rules to whatever is advised. I cannot change the default logs unfortunately.

Reply:
Thanks for the clarification.

I have forwarded those examples to the Log team and they will build an integration pipeline that will be automatically installed when the logs are sent with the source `contrastsecurity`.

So no need to add the part about pipeline creation in the documentation.